client/llb/source.go

```diff
@@ -452,6 +453,45 @@ func Differ(t DiffType, required bool) LocalOption {
 	})
 }
 
+func OCILayout(contentStoreID, digest string, opts ...OCILayoutOption) State {
+	gi := &OCILayoutInfo{}
+
+	for _, o := range opts {
+		o.SetOCILayoutOption(gi)
+	}
+	attrs := map[string]string{}
+	if gi.SessionID != "" {
+		attrs[pb.AttrOCISessionID] = gi.SessionID
+		addCap(&gi.Constraints, pb.CapSourceOCILayoutSessionID)
+	}
+
+	addCap(&gi.Constraints, pb.CapSourceOCILayout)
+
+	source := NewSource(fmt.Sprintf("oci-layout://%s@%s"+contentStoreID, digest), attrs, gi.Constraints)
+	return NewState(source.Output())
+}
+
+type OCILayoutOption interface {
```

Comment:
I don't see a way to pass layer limit in this mode. Maybe it is possible to share the `ImageOptions` in here. Not all imageoptions are supported in here and `SessionID` is not supported for regular images. But some are the same(layerlimit etc) and want to make sure we don't drop any.

Reply:
If they don't quite overlap, then I can copy them over. It doesn't hurt. 

What others are relevant? ImageInfo looks like:

```go
type ImageInfo struct {
	constraintsWrapper
	metaResolver  ImageMetaResolver
	resolveDigest bool
	resolveMode   ResolveMode
	layerLimit    *int
	RecordType    string
}
```

We already have `constraintsWrapper` (as all do) and now `layerLimit`. Which others?